server/src/main/java/org/elasticsearch/index/mapper/CompositeRuntimeField.java

```diff
@@ -118,11 +124,6 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         for (FieldMapper.Parameter<?> parameter : parameters) {
             parameter.toXContent(builder, includeDefaults);
         }
-        builder.startObject("fields");
-        for (RuntimeField subfield : subfields) {
-            subfield.toXContent(builder, params);
-        }
-        builder.endObject();
```

Comment:
don't we need to print out the subfields? Shouldn't the toXContent output be the same as what users submit? Do we have tests for the toXContent output? I am a bit confused about what this causes etc. can you expand?

Reply:
We previously needed to print these out separately because the `fields` parameter would get emptied out after parsing.  But now that we take a copy of it for parsing, having this extra bit of code in here results in the fields being emitted twice, which causes test failures all over the place.  There are existing tests for xcontent serialization, which is what caught this.

Reply:
The loop above prints the parameters, one of which is `fields`.  Before this commit it contained an empty map, so it didn't print anything.  Now the map doesn't get emptied and so things get printed.

Reply:
I see!!!